build-self-coverage.js

```diff
@@ -1,21 +1,42 @@
-var istanbul = require('istanbul-lib-instrument')
-var fs = require('fs')
-var path = require('path')
-
-;[
-  'index.js',
-  'lib/process.js'
-].forEach(function (name) {
-  var indexPath = path.join(__dirname, name)
-  var source = fs.readFileSync(indexPath, 'utf8')
-
-  var instrumentor = istanbul.createInstrumenter({
-    coverageVariable: '___NYC_SELF_COVERAGE___',
-    esModules: true
-  })
+const path = require('path')
+const fs = require('fs')
+const istanbul = require('istanbul-lib-instrument')
+const makeDir = require('make-dir')
+const glob = require('glob')
+
+const instrumenter = istanbul.createInstrumenter({
+  coverageVariable: '___NYC_SELF_COVERAGE___',
+  esModules: true
+})
+
+function instrumentFile (name) {
+  const indexPath = path.join(__dirname, name)
+  const outputPath = path.join(__dirname, 'self-coverage', name)
 
-  var instrumentedSource = instrumentor.instrumentSync(source, indexPath)
+  const source = fs.readFileSync(indexPath, 'utf8')
+  const instrumentedSource = name === 'package.json' ? source : instrumenter.instrumentSync(source, indexPath)
 
-  var outputPath = path.join(__dirname, name.replace(/\.js$/, '.covered.js'))
+  makeDir.sync(path.dirname(outputPath))
   fs.writeFileSync(outputPath, instrumentedSource)
-})
+}
+
+function instrumentGlob (pattern) {
+  const result = glob.sync(pattern, {
+    cwd: __dirname,
+    nodir: true
+  })
+
+  result.forEach(file => {
+    instrumentFile(file)
+  })
+}
+
+function instrumentAll () {
+  const globPatterns = ['package.json', 'index.js', 'bin/*.js', 'lib/**/*.js']
```

Comment:
should we be instrumenting `package.json`?

Reply:
We're not, I added a comment.